File: leetcode1695/leetcode1695.py
from collections import Counter, defaultdict


class Solution:
    def maximumUniqueSubarray(self, nums: List[int]) -> int:
        ans = 0
        left = 0
        _sum = 0
        cnt = defaultdict(int)
        for right in range(len(nums)):
            _sum += nums[right]
            cnt[nums[right]] += 1
            while cnt[nums[right]] > 1:
                _sum -= nums[left]
                cnt[nums[left]] -= 1
                if cnt[nums[left]] == 0:
                    cnt.pop(nums[left])
                left += 1
            ans = max(ans, _sum)
        return ans

File: leetcode1695/leetcode1695.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
    public:
        int maximumUniqueSubarray(vector<int>& nums) {
            int ans = 0, left = 0, sum = 0;
            unordered_map<int, int> cnt;
            for (int right = 0; right < nums.size(); right ++){
                sum += nums[right];
                cnt[nums[right]] ++;
                while (cnt[nums[right]] > 1)
                {
                    sum -= nums[left];
                    cnt[nums[left]]--;
                    if (cnt[nums[left]] == 0) cnt.erase(nums[left]);
                    left ++;
                }
                ans = max(ans, sum);
            }
            return ans;
        }
    };